src/builtins/BuiltinTemporal.cpp

```diff
@@ -491,6 +496,239 @@ static Value builtinTemporalPlainDateTimeInLeapYear(ExecutionState& state, Value
     return TemporalCalendarObject::calendarInLeapYear(state, thisValue.asObject()->asTemporalPlainDateTimeObject()->getCalendar().asObject(), thisValue);
 }
 
+static Value builtinTemporalZonedDateTimeConstructor(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    CHECK_TEMPORAL_ZONED_DATE_TIME(state, thisValue);
+
+    if (!argv[0].isBigInt()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Invalid type");
+    }
+
+    if (!TemporalInstantObject::isValidEpochNanoseconds(argv[0])) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, "epoch Nanoseconds is out of range");
+    }
+
+    auto timeZone = TemporalTimeZoneObject::toTemporalTimeZone(state, argv[1]).asObject()->asTemporalTimeZoneObject();
+    auto calendar = TemporalCalendarObject::toTemporalCalendarWithISODefault(state, argc > 2 ? argv[2] : Value()).asObject()->asTemporalCalendarObject();
```

Comment:
`Value().asObject()` is working? It should be undefined, not an object.

Reply:
The Value() is a parameter of the function and the function returned value is an object.